kornia/augmentation/container/augment.py

```diff
@@ -368,74 +340,72 @@ def forward(  # type: ignore[override]
-            if isinstance(arg, (Boxes,)):
-                input = arg.data  # all boxes are in (B, N, 4, 2) format now.
-            else:
-                input = arg
-
-            for param in params:
-                module = self.get_submodule(param.name)
-                if (
-                    isinstance(module, IntensityAugmentationBase2D)
-                    and dcate in DataKey
-                    and not isinstance(module, RandomErasing)
-                ):
-                    pass  # Do nothing
-                elif isinstance(module, ImageSequential) and module.is_intensity_only() and dcate in DataKey:
-                    pass  # Do nothing
-                elif isinstance(module, VideoSequential) and dcate not in [DataKey.INPUT, DataKey.MASK]:
-                    batch_size: int = input.size(0)
-                    input = input.view(-1, *input.shape[2:])
-                    input, label = ApplyInverse.apply_by_key(input, label, module, param, dcate, extra_args=extra_args)
-                    input = input.view(batch_size, -1, *input.shape[1:])
-                elif isinstance(module, PatchSequential):
-                    raise NotImplementedError("Geometric involved PatchSequential is not supported.")
-                elif (
-                    isinstance(module, (GeometricAugmentationBase2D, ImageSequential, RandomErasing))
-                    and dcate in DataKey
-                ):
-                    input, label = ApplyInverse.apply_by_key(input, label, module, param, dcate, extra_args=extra_args)
-                elif isinstance(module, MixAugmentationBaseV2):
-                    if dcate in [DataKey.BBOX_XYXY, DataKey.BBOX_XYWH]:
-                        dcate = DataKey.BBOX
-                    input = module(input, params=param.data, data_keys=[dcate])
-                elif isinstance(module, (SequentialBase,)):
-                    raise ValueError(f"Unsupported Sequential {module}.")
-                else:
-                    raise NotImplementedError(f"data_key {dcate} is not implemented for {module}.")
+        outputs: Union[Tensor, List[DataType]] = in_args
+        for param in params:
+            module = self.get_submodule(param.name)
+            outputs = self.transform_op.transform(*outputs, module=module, param=param, extra_args=self.extra_args)
+            if not isinstance(outputs, (list, tuple)):
+                # Make sure we are unpacking a list whilst post-proc
+                outputs = [outputs]
+            self._update_transform_matrix_by_module(module)
+
+        outputs = self._arguments_postproc(args, outputs, data_keys=self.transform_op.data_keys)
+        # Restore it back
+        self.transform_op.data_keys = self.data_keys
+
+        self._params = params
+
+        if len(outputs) == 1 and isinstance(outputs, list):
+            return outputs[0]
+
+        return outputs
+
+    def _preproc_boxes(self, arg: DataType, dcate: DataKey) -> Boxes:
```

Comment:
I will try to review it a bit more in-depth, at least the parts I'm more familiar with. But this is removing support for lists of boxes. Of course, wrapping with the `Boxes` class allows to do that easily but if this is what you want to push there should be some clear message that lists are not being supported anymore and pointing to the solution.